controllers/webhook.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2025 Red Hat OpenShift Data Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	admrv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	webhookServiceName = "odf-operator-webhook-service"
+	webhookName        = "csv.odf.openshift.io"
+	WebhookPath        = "/mutate-operators-coreos-com-v1alpha1-csv"
+)
+
+var (
+	webhookService = corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookServiceName,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "https",
+					Port:       443,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt32(9443),
+				},
+			},
+			Selector: map[string]string{
+				"app.kubernetes.io/name": "odf-operator",
+			},
+			Type: corev1.ServiceTypeClusterIP,
+		},
+	}
```

Comment:
You are only interested in the spec, there is no point in having a complete object here
```suggestion
	webhookService = corev1.ServiceSpec{
		Ports: []corev1.ServicePort{
			{
				Name:       "https",
				Port:       443,
				Protocol:   corev1.ProtocolTCP,
				TargetPort: intstr.FromInt32(9443),
			},
		},
		Selector: map[string]string{
			"app.kubernetes.io/name": "odf-operator",
		},
		Type: corev1.ServiceTypeClusterIP,
	}
```

Reply:
No i would like to have it as a whole, I removed the constants.